Source/Diagnostics/ParticleDiag/ParticleDiag.cpp

```diff
@@ -28,12 +30,17 @@ ParticleDiag::ParticleDiag(
     amrex::Vector<std::string> variables;
     const int variables_specified = pp_diag_name_species_name.queryarr("variables", variables);
 
-    if (variables_specified){
+    if (variables_specified) {
         // If only specific variables have been specified, fill m_plot_flags with zero and only set
         // requested variables to one
         std::fill(m_plot_flags.begin(), m_plot_flags.end(), 0);
+        bool contains_positions = false;
         if (variables[0] != "none"){
-            const std::map<std::string, int> existing_variable_names = pc->getParticleComps();
+            std::map<std::string, int> existing_variable_names = pc->getParticleComps();
+#ifdef WARPX_DIM_RZ
+            // we reconstruct to Cartesian x,y,z for RZ particle output
+            existing_variable_names["y"] = PIdx::theta;
```

Comment:
This is a hack and was already broken before.
#4917 will do a general overhaul of all this logic and remove it.